### PR TITLE
APTT-15 Disabled measurements collection when no active metric exists

### PR DESCRIPTION
**Pull request by SwinX**
Prevented measurements creation in _RPTracker for all measurement creation methods.
Ensured that metric is still created when collection of non-metric measurements disabled.
Added Kiwi BDD framework to the project, implemented unit-tests for new feature in BDD style.
Slightly fixed _RPTrackingManager configuration tests.

### CODE REVIEW

**Comment by SwinX** [comment]
@donnie-jp @vito1188 

2 points to mention: 
- Tracking of non-metric measurements is enabled by default in `_RPTTracker`
- If in config payload missing `enableNonMetricMeasurement` field, non-metric measurements still will be enabled. In this case config will not be discarded because this fails some of our integration tests checking that something works on config from prod. (never got into much detail about this). Also designing in this way will not block releasing this feature while it's still missing on prod.

File: RPerformanceTracking.xcodeproj/project.pbxproj [hunk 1]
```diff
@@ -7,6 +7,9 @@
 	objects = {
 
 /* Begin PBXBuildFile section */
+		1696C3CA207714D700288BB5 /* RPTConfigurationTests.m in Sources */ = {isa = PBXBuildFile; fileRef = 1696C3C9207714D700288BB5 /* RPTConfigurationTests.m */; };
+		1696C3CD2077157200288BB5 /* TestUtils.m in Sources */ = {isa = PBXBuildFile; fileRef = 1696C3CC2077157200288BB5 /* TestUtils.m */; };
+		3059D7739CD8B79D17CD17F0 /* Pods_Tests.framework in Frameworks */ = {isa = PBXBuildFile; fileRef = EDB01895B3BE708F72F5A39E /* Pods_Tests.framework */; };
 		5A8CF0CD1E9C76A1006E7544 /* UIControlTests.m in Sources */ = {isa = PBXBuildFile; fileRef = 5A8CF0CC1E9C76A1006E7544 /* UIControlTests.m */; };
```
Comment (donnie-jp) [inline RPerformanceTracking.xcodeproj/project.pbxproj]: We don't commit the CocoaPods integrated version of the project file.
It's cleaner that way and easier to see real project file changes in the diff.
Please run `pod deintegrate` before committing.

File: RPerformanceTracking/Private/_RPTConfiguration.m [hunk 2]
```diff
@@ -30,6 +29,12 @@ - (instancetype)initWithData:(NSData *)data
 
         NSURL *url = [NSURL URLWithString:urlString];
         if (!url) break;
+        
+        NSString* enableNonMetricMeasurement = values[@"enableNonMetricMeasurement"];
```
Comment (donnie-jp) [inline RPerformanceTracking/Private/_RPTConfiguration.m]: not NSNumber?

File: Tests/TrackerTests.m [hunk 3]
```diff
@@ -6,6 +6,124 @@
...
+#import <Underscore_m/Underscore.h>
+
+#import "TestUtils.h"
+
+#pragma clang diagnostic push
+#pragma clang diagnostic ignored "-Wnonnull"
+
+SPEC_BEGIN(RPTTRackerTests)
+
+describe(@"RPTTracker", ^{
```
Comment (donnie-jp) [inline Tests/TrackerTests.m:19]: Nice tests!

**Review by donnie-jp — APPROVED** [review]

File: RPerformanceTracking/Private/_RPTConfiguration.m (hunk 2 above)
Comment (SwinX) [reply RPerformanceTracking/Private/_RPTConfiguration.m]: Yes, this should be `NSNumber`

**Pull request merged**